tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/caches/AbstractCache.scala

```diff
@@ -55,24 +55,37 @@ abstract class AbstractCache[ContextT, SourceT, TargetT]
     } else {
       // introduce a new constant
       val (newContext, targetValue) = create(context, srcValue)
-      cache = cache + (srcValue -> (targetValue, level))
-      reverseCache = reverseCache + (targetValue -> (srcValue, level))
+      addToCache(srcValue, targetValue)
+      cache += srcValue -> (targetValue, level)
+      reverseCache += targetValue -> (srcValue, level)
       (newContext, targetValue)
     }
   }
 
+  /**
+   * Add a value to the cache. Implementations of this class may use this method for a more efficient mass construction
```

Comment:
```suggestion
   * Add a value to the cache at the current level. Implementations of this class may use this method for a more efficient mass construction
```

Reply:
I'm also not sure I understand the second sentence. Nothing about this method per-se seems to make things more efficient?

Reply:
It does call `distinct` over `n` arguments once, instead of introducing `n` calls to `distinct`.

Reply:
I will add a comment

Reply:
Right. I have removed that other sentence. It is confusing.